lib/absinthe/phase/document/arguments/variable_types_match.ex

```diff
@@ -47,12 +49,67 @@ defmodule Absinthe.Phase.Document.Arguments.VariableTypesMatch do
 
   def check_variable_types(%Operation{} = op) do
     variable_defs = Map.new(op.variable_definitions, &{&1.name, &1})
+
+    # This is the new/breaking implementation which will now only log warnings
+    # for type mismatches.
     Blueprint.prewalk(op, &check_variable_type(&1, op.name, variable_defs))
+
+    # This is the legacy implementation that we will continue to use until
+    # everything is patched.
+    Blueprint.prewalk(op, &check_var_type(&1, op.name, variable_defs))
   end
 
   def check_variable_types(%Operation{} = op, %Fragment.Named{} = fragment) do
     variable_defs = Map.new(op.variable_definitions, &{&1.name, &1})
+
+    # This is the new/breaking implementation which will now only log warnings
+    # for type mismatches.
     Blueprint.prewalk(fragment, &check_variable_type(&1, op.name, variable_defs))
+
+    # This is the legacy implementation that we will continue to use until
+    # everything is patched.
+    Blueprint.prewalk(fragment, &check_var_type(&1, op.name, variable_defs))
+  end
+
+  defp check_var_type(%{schema_node: nil} = node, _, _) do
```

Comment:
All `check_var_type` functions were copied from the older revision of this file -- preserving old behavior. We only run the new flow (`check_variable_type`) to log a warning to the user.